### ISSUE TITLE
"set term qt: unknown or ambiguous terminal type" when running test suite

### ISSUE THREAD

**Issue by apjanke** [octave-app/octave-app-bundler#18]
When running the test suite for 4.4.0 beta2 in the GUI, I see a bunch of these warnings:

```
  plot/util/hgsave.m ..........................................
set term qt
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization
...
set term qt
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

  ./base/gsicc_manage.c:1172: gsicc_open_search(): Could not find default_rgb.icc
| ./base/gsicc_manage.c:1838: gsicc_set_device_profile(): cannot find device profile

set term qt
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

  ./base/gsicc_manage.c:1172: gsicc_open_search(): Could not find default_rgb.icc
| ./base/gsicc_manage.c:1838: gsicc_set_device_profile(): cannot find device profile
```

**Comment by apjanke** [comment]
Based on https://github.com/Homebrew/homebrew-science/issues/1275, it sounds like that error happens if `gnuplot` was built without Qt bindings. And indeed, in the `gnuplot` formula, the `qt` dependency is `:optional`, not `:recommended`.

Meanwhile, the GUI for my Octave.app build is not plotting anything.

```
>> sombrero

set term qt
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization


gnuplot> set terminal aqua enhanced title "Figure 1" size 560 420  font "*,6.66667" dashlength 1
                      ^
         line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Plotting with an 'unknown' terminal.
No output will be generated. Please select a terminal with 'set terminal'.
```

I'm going to try customizing the `gnuplot` formula to install the Qt bindings.

**Comment by apjanke** [comment]
That was at least part of it. Building `gnuplot` with Qt bindings made plotting work again, and I think I'm seeing fewer of the "unknown or ambiguous terminal type" errors in the test suite. But some of them are still there.

```
  publish.tst .................................................
set term qt
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

  ./base/gsicc_manage.c:1172: gsicc_open_search(): Could not find default_rgb.icc
| ./base/gsicc_manage.c:1838: gsicc_set_device_profile(): cannot find device profile

set term qt
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

  ./base/gsicc_manage.c:1172: gsicc_open_search(): Could not find default_rgb.icc
| ./base/gsicc_manage.c:1838: gsicc_set_device_profile(): cannot find device profile

set term qt
         ^
line 0: unknown or ambiguous terminal type; type just 'set terminal' for a list

WARNING: Error during initialization

  ./base/gsicc_manage.c:1172: gsicc_open_search(): Could not find default_rgb.icc
| ./base/gsicc_manage.c:1838: gsicc_set_device_profile(): cannot find device profile
...
```

Made a `gnuplot-octave-app` formula for the customizations. Enabled Cairo while I was in there, since I had seen Schoeps do that in his formulae, I think.

Fewer failing tests; we're down to 4.

```
Summary:

  PASS                            14933
  FAIL                                4
```

**Comment by apjanke** [comment]
I reinstalled my main Homebrew's `gnuplot --with-qt --with-cairo` to see if that would make the errors in the test suite go away. Suspected maybe Octave was calling out to a default /usr/local/bin/gnuplot for some of its work.

They didn't go away. That wasn't it.

**Comment by apjanke** [comment]
This is one of the tests that triggered it. I determined this by doing `test legend verbose`.

```
***** testif ; any (strcmp ("gnuplot", available_graphics_toolkits ()))
 toolkit = graphics_toolkit ("gnuplot");
 h = figure ("visible", "off");
 unwind_protect
   position = get (h, "position");
   plot (rand (3));
   legend ();
   filename = sprintf ("%s.eps", tempname ());
   print (filename);
   unlink (filename);
   assert (get (h, "position"), position);
 unwind_protect_cleanup
   close (h);
   graphics_toolkit (toolkit);
 end_unwind_protect
```

Maybe using the plain "gnuplot" graphics toolkit is an issue?

**Comment by apjanke** [comment]
Aha, that was the problem: a bad `gnuplot-octave-app` build. When I changed the dependency from `qt` to `qt-octave-app`, the implicit `--with-qt` option went away, and I needed to add in an explicit `option --without-qt` to make the `build.with?("qt")` tests work right. (Since I'm still just calling the option "qt" instead of renaming the tests and option.)

Fixed in https://github.com/octave-app/homebrew-octave-app-bases/commit/9d8616f83e53542ad990b253badcf0d7cb333be1.

Rebuilt Octave.app with a fixed gnuplot-octave-app, and those errors went away.

**Comment by apjanke** [comment]
This problem came back in the 4.4.1 RC1 build. We switched to using an un-hacked `qt` formula, and that had the side effect of restoring the `--with-qt` option and making it non-default. So its newly-grabbed gnuplot formula was building without Qt.

I modified our gnuplot_5.2.6 formula to build with Qt in https://github.com/octave-app/homebrew-octave-app/commit/fcbbd5997cf1e7146ed52420e84dc01a0b73d714
